### DISCUSSION TITLE
Fabric Fast: Billing Account creation and IAM bindings?

### DISCUSSION THREAD

**Discussion by justinphebey** [GoogleCloudPlatform/cloud-foundation-fabric#2626] [Q&A] [answered]
Hi

I'm getting started with a new organisation (in GCP) and have been reading the Medium article by Darren Lester that elaborates on the Fabric Fast process for the first couple of stages. This recommends someone only follow the first two steps in the cloud foundation guided steps that appear in the GCP cloud console.  

Does Fabric Fast also require that I should set up a billing account if I'm setting up a new organisation and don't already have any billing account, before I use the 0-bootstrap blueprint?

The readme in fast/stages/0-bootstrap also mentions a couple of recommended options for whether the IAM bindings should be set at the organisation level or on the billing account itself. Are there any negative implications of choosing these to be bound at the organisation level rather than more specifically at the billing account level? Could you please provide some additional context as to why someone would choose each option?

Re: 
https://medium.com/google-cloud/google-cloud-landing-zone-with-terraform-and-cloud-foundation-fabric-fast-part-1-2c7634ac31fd
https://medium.com/google-cloud/google-cloud-landing-zone-with-terraform-and-cloud-foundation-fabric-fast-part-2-6bd067f5bed7

**Comment by juliocc** [comment, marked as answer]
> Does Fabric Fast also require that I should set up a billing account if I'm setting up a new organisation and don't already have any billing account, before I use the 0-bootstrap blueprint?

A billing account is required, it doesn't matter where it comes from. If you already have a billing account, you can use it.

> Are there any negative implications of choosing these to be bound at the organisation level rather than more specifically at the billing account level? Could you please provide some additional context as to why someone would choose each option?

It's not a matter of better/worse alternatives but rather how organizations and billing accounts are structured. Some users have a single organization and a single billing account, but there are more complex scenarios with multiple organizations that leverage a single billing account. FAST can accomodate for both of these cases (and others too).

The bottom line is that you need a billing account where you are Billing Account Administrator. Whether that billing account is part of your organization or not has little impact except on how permissions are managed. FAST tries to hide this complexity and that's why it allows you to pass in different configurations depending on your situation.

**Reply by justinphebey** [reply]
Thanks Julio,

I wasn't sure if not supplying a billing account id to the terraform stage would automatically generate a fresh one. I will create one of these via cloud console now, prior to starting with the terraform stages.

If I've understood you correctly regarding multiple billing accounts, there aren't any cost analysis advantages to using multiple? I'm assuming therefore that labels can provide granular analysis via a single billing account?

I did also find this regarding multiple billing accounts in case anyone else stumbles across this question, so will run this past my accountant too.

https://cloud.google.com/billing/docs/onboarding-checklist#cloud-billing-accounts

Kind regards
Justin

**Reply by juliocc** [reply]
> If I've understood you correctly regarding multiple billing accounts, there aren't any cost analysis advantages to using multiple? 

As usual, it depends. For some organizations it might make sense to have multiple billing accounts. But in general a single account is fine and simpler to manage.

> I'm assuming therefore that labels can provide granular analysis via a single billing account?

Exactly. Labels can help you understand how your spending is happening. You can also enable the [billing export](https://cloud.google.com/billing/docs/how-to/export-data-bigquery) to get a lot more details about all the charges happening to your account.
